src/main/java/edu/wisc/my/personalizedredirection/controller/PersonalizedRedirectionController.java

```diff
@@ -79,7 +82,13 @@ public void setRedirectionService(IRedirectionService redirectionService) {
             logger.error("Issues happened while trying to generate custom link",
                     e);
             response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            hasError = true;
          }
+
+         if(hasError){
+            response.sendRedirect(errorURL);
```

Comment:
The redirect to /`context-name`/404.html doesn't work because this method catches the 404 and treats it like the `appName`